### PR TITLE
Fixing tutorial to use DatagenConnector vs. Voluable

### PR DESCRIPTION
**Pull request by bbejeck**
### Description

From [asana 1 ](https://app.asana.com/0/1201906632362444/1204950994774439/f), [asana 2](https://app.asana.com/0/1201906632362444/1204950936311544/f), and [asana 3](https://app.asana.com/0/1201906632362444/1205067366905183/f)

This PR also includes changes for the semaphore deployment files to work without the cache.

### Staging Docs

http://kafka-tutorials-staging.s3-website-us-west-2.amazonaws.com/MINOR_address_userback_issues/kafka-console-consumer-primitive-keys-values/kafka.html

http://kafka-tutorials-staging.s3-website-us-west-2.amazonaws.com/MINOR_address_userback_issues/generate-streams-of-test-data/ksql.html

http://kafka-tutorials-staging.s3-website-us-west-2.amazonaws.com/MINOR_address_userback_issues/generate-streams-of-test-data/kafka.html

### CODE REVIEW

**Comment by bbejeck** [comment]
test for the [updated tutorial passed](https://semaphore.ci.confluent.io/jobs/bb2e4369-9d36-4b9b-8436-b1b07e0e7ad8)

**Comment by bbejeck** [comment]
Failures unrelated - [https://semaphore.ci.confluent.io/jobs/21adb449-41a9-443e-be66-765953839680](tests for Kafka Generate streams of test data) passes

**Comment by bbejeck** [comment]
Failures unrelated test for [ksql generate test streams of data](https://semaphore.ci.confluent.io/jobs/9f183585-f125-410a-9da2-bd83957ef222) passed

**Comment by bbejeck** [comment]
@davetroiano this is ready for review

File: _includes/tutorials/generate-test-data-streams/kafka/markup/dev/create-connector-02.adoc [hunk 1]
```diff
@@ -1,46 +1,34 @@
...
-'attrkp.traffic.null.rate'           = 1,
-----
-
--> The MAC address of the device is set as one of the existing keys on `devices` 
++++++
+<pre class="snippet"><code class="sql">{% include_raw tutorials/generate-test-data-streams/kafka/code/tutorial-steps/dev/create-connector-03.sh %}</code></pre>
++++++
 
--> The rate at which messages are produced is limited to two per second (`'throttle.ms' = 500`)
+When looking at the connector setting notice the `quickstart` entry in the `JSON`, this is where you specify a bundle schema the datagen connector provides.  By generating two streams of data with related keys you'll be able to join records or perform aggregations with the related keys.
```
Comment (davetroiano) [inline _includes/tutorials/generate-test-data-streams/kafka/markup/dev/create-connector-02.adoc:33]: bundle schema -> bundled schema

File: _includes/tutorials/generate-test-data-streams/ksql/code/src/statements.sql [hunk 2]
```diff
@@ -1,26 +1,38 @@
...
-    'connector.class'             = 'io.mdrogalis.voluble.VolubleSourceConnector',
-    'key.converter'               = 'org.apache.kafka.connect.storage.StringConverter',
-    'genkp.clicks.with'           = '#{Number.randomDigit}',
-    'attrkp.clicks.null.rate'     = 1,
-    'genv.clicks.source_ip.with'  = '#{Internet.ipV4Address}',
-    'genv.clicks.host.with'       = '#{Internet.url}',
-    'genv.clicks.path.with'       = '#{File.fileName}',
-    'genv.clicks.user_agent.with' = '#{Internet.userAgentAny}',
-    'topic.clicks.throttle.ms'    = 1000 
+CREATE SOURCE CONNECTOR IF NOT EXISTS PIZZA_ORDERS WITH (
```
Comment (davetroiano) [inline _includes/tutorials/generate-test-data-streams/ksql/code/src/statements.sql:1]: suggest aligning all the `=`'s for readability here and in the two statements below

File: _includes/tutorials/generate-test-data-streams/ksql/code/tutorial-steps/dev/create-connector-01.sql [hunk 3]
```diff
@@ -1,11 +1,12 @@
...
-    'key.converter'               = 'org.apache.kafka.connect.storage.StringConverter',
-    'genkp.clicks.with'           = '#{Number.randomDigit}',
-    'attrkp.clicks.null.rate'     = 1,
-    'genv.clicks.source_ip.with'  = '#{Internet.ipV4Address}',
-    'genv.clicks.host.with'       = '#{Internet.url}',
-    'genv.clicks.path.with'       = '#{File.fileName}',
-    'genv.clicks.user_agent.with' = '#{Internet.userAgentAny}',
-    'topic.clicks.throttle.ms'    = 1000 
-);
+CREATE SOURCE CONNECTOR IF NOT EXISTS PIZZA_ORDERS WITH (
```
Comment (davetroiano) [inline _includes/tutorials/generate-test-data-streams/ksql/code/tutorial-steps/dev/create-connector-01.sql:1]: suggest aligning all the `=`'s for readability

File: _includes/tutorials/generate-test-data-streams/ksql/code/tutorial-steps/dev/check-connector-01_expected.log [hunk 4]
```diff
@@ -1,4 +1,4 @@
- Connector Name | Type   | Class                                       | Status
------------------------------------------------------------------------------------------------------
- CLICKS         | SOURCE | io.mdrogalis.voluble.VolubleSourceConnector | RUNNING (1/1 tasks RUNNING)
------------------------------------------------------------------------------------------------------
\ No newline at end of file
+Connector Name | Type   | Class                                               | Status
```
Comment (davetroiano) [inline _includes/tutorials/generate-test-data-streams/ksql/code/tutorial-steps/dev/check-connector-01_expected.log:1]: add space to align the `|` characters

File: _includes/tutorials/generate-test-data-streams/ksql/code/tutorial-steps/dev/check-connector-02_expected.log [hunk 5]
```diff
@@ -1,5 +1,6 @@
- Connector Name | Type   | Class                                       | Status
------------------------------------------------------------------------------------------------------
- CLICKS          | SOURCE | io.mdrogalis.voluble.VolubleSourceConnector | RUNNING (1/1 tasks RUNNING)
- NETWORK_TRAFFIC | SOURCE | io.mdrogalis.voluble.VolubleSourceConnector | RUNNING (1/1 tasks RUNNING)
------------------------------------------------------------------------------------------------------
\ No newline at end of file
+Connector Name | Type   | Class                                               | Status
```
Comment (davetroiano) [inline _includes/tutorials/generate-test-data-streams/ksql/code/tutorial-steps/dev/check-connector-02_expected.log:1]: add space to align the `|` character

File: _includes/tutorials/generate-test-data-streams/ksql/code/tutorial-steps/dev/create-connector-02.sql [hunk 6]
```diff
@@ -1,14 +1,12 @@
...
-    'genv.devices.name.with'              = '#{GameOfThrones.dragon}',
-    'genv.devices.location->city.with'    = '#{Address.city}',
-    'genv.devices.location->country.with' = '#{Address.country}',
-    'topic.devices.records.exactly'       = 10,
-    'genkp.traffic.with'                  = '#{Number.randomDigit}',
-    'attrkp.traffic.null.rate'            = 1,
-    'genv.traffic.mac.matching'           = 'devices.key',
-    'genv.traffic.bytes_sent.with'        = '#{Number.numberBetween ''64'',''4096''}',
-    'topic.traffic.throttle.ms'           = 500
+CREATE SOURCE CONNECTOR IF NOT EXISTS FLEET_LOCATION WITH (
```
Comment (davetroiano) [inline _includes/tutorials/generate-test-data-streams/ksql/code/tutorial-steps/dev/create-connector-02.sql:1]: align the `=`'s

File: _includes/tutorials/generate-test-data-streams/ksql/code/tutorial-steps/dev/create-connector-03.sql [hunk 7]
```diff
@@ -0,0 +1,12 @@
+CREATE SOURCE CONNECTOR IF NOT EXISTS FLEET_SENSORS WITH (
```
Comment (davetroiano) [inline _includes/tutorials/generate-test-data-streams/ksql/code/tutorial-steps/dev/create-connector-03.sql:1]: align the `=`'s

File: _includes/tutorials/generate-test-data-streams/ksql/markup/dev/create-connector-02.adoc [hunk 8]
```diff
@@ -1,46 +1,31 @@
-Now let's see how you can generate streams of data that is related and consistent. 
+Now let's see how you can generate streams of data that is related and consistent. You are going to use https://github.com/confluentinc/kafka-connect-datagen/tree/master#use-a-bundled-schema-specification[bundled schemas] available in the Kafka Connect `DatagenConnector`.
 
-Consider an example where we want to generate some dummy data about network devices. The stream of events would have fields such as
 
--> The MAC address of the device
+Consider an example where we want to generate some dummy data for tracking sensors on a fleet of delivery trucks.  There will be two streams of data,
```
Comment (davetroiano) [inline _includes/tutorials/generate-test-data-streams/ksql/markup/dev/create-connector-02.adoc:4]: second sentence is missed the second stream? suggest:

There will be two streams of data: (1) the current location of each truck, and (2) truck sensor information from each truck.

File: _includes/tutorials/generate-test-data-streams/ksql/markup/dev/create-connector-02.adoc [hunk 9]
```diff
@@ -1,46 +1,31 @@
...
-* **Devices**
-
-->  `genkp` sets the https://dius.github.io/java-faker/apidocs/com/github/javafaker/Internet.html#macAddress--[macAddress] as the primitive key
-
--> The `city` and `country` are set as nested values in the `location` entity
-
--> `records.exactly` limits the number of records produced to 10
-
-* **Traffic**
+Then create the fleet management sensor reading connector you'll execute the following:
```
Comment (davetroiano) [inline _includes/tutorials/generate-test-data-streams/ksql/markup/dev/create-connector-02.adoc:26]: "to" create the fleet .... you'll execute the following

File: _includes/tutorials/generate-test-data-streams/kafka/code/tutorial-steps/dev/create-connector-01.sh [hunk 10]
```diff
@@ -1,13 +1,15 @@
-curl -X PUT http://localhost:8083/connectors/clicks/config \
+curl -X PUT http://localhost:8083/connectors/pizza_orders/config \
      -i -H "Content-Type: application/json" -d '{
-    "connector.class"             : "io.mdrogalis.voluble.VolubleSourceConnector",
-    "key.converter"               : "org.apache.kafka.connect.storage.StringConverter",
+    "connector.class"             : "io.confluent.kafka.connect.datagen.DatagenConnector",
```
Comment (davetroiano) [inline _includes/tutorials/generate-test-data-streams/kafka/code/tutorial-steps/dev/create-connector-01.sh:3]: suggest aligning all the `:`'s for readability

File: _includes/tutorials/generate-test-data-streams/kafka/code/tutorial-steps/dev/create-connector-02.sh [hunk 11]
```diff
@@ -1,15 +1,13 @@
...
-     "genv.devices.name.with"              : "#{GameOfThrones.dragon}",
-     "genv.devices.location->city.with"    : "#{Address.city}",
-     "genv.devices.location->country.with" : "#{Address.country}",
-    "topic.devices.records.exactly"        : 10,
-     "genkp.traffic.with"                : "#{Number.randomDigit}",
-    "attrkp.traffic.null.rate"           : 1,
-      "genv.traffic.mac.matching"        : "devices.key",
-      "genv.traffic.bytes_sent.with"     : "#{Number.numberBetween \u002764\u0027,\u00274096\u0027}",
-     "topic.traffic.throttle.ms"         : 500 
+     "connector.class"             : "io.confluent.kafka.connect.datagen.DatagenConnector",
```
Comment (davetroiano) [inline _includes/tutorials/generate-test-data-streams/kafka/code/tutorial-steps/dev/create-connector-02.sh:3]: suggest aligning all the `:`'s for readability

File: _includes/tutorials/generate-test-data-streams/kafka/code/tutorial-steps/dev/create-connector-03.sh [hunk 12]
```diff
@@ -0,0 +1,13 @@
+curl -X PUT http://localhost:8083/connectors/fleet_sensor/config \
```
Comment (davetroiano) [inline _includes/tutorials/generate-test-data-streams/kafka/code/tutorial-steps/dev/create-connector-03.sh:1]: align the `:`'s

File: _includes/tutorials/generate-test-data-streams/kafka/markup/dev/create-connector-02.adoc [hunk 13]
```diff
@@ -1,46 +1,34 @@
-Now let's see how you can generate streams of data that is related and consistent. 
+Now let's see how you can generate streams of data that is related and consistent. You are going to use https://github.com/confluentinc/kafka-connect-datagen/tree/master#use-a-bundled-schema-specification[bundled schemas] available in the Kafka Connect `DatagenConnector`.
```
Comment (davetroiano) [inline _includes/tutorials/generate-test-data-streams/kafka/markup/dev/create-connector-02.adoc:1]: ... streams of data that are related and consistent.

File: _includes/tutorials/generate-test-data-streams/kafka/markup/dev/create-connector-02.adoc [hunk 14]
```diff
@@ -1,46 +1,34 @@
...
 
-Consider an example where we want to generate some dummy data about network devices. The stream of events would have fields such as
 
--> The MAC address of the device
+Consider an example where we want to generate some dummy data for tracking sensors on a fleet of delivery trucks.  There will be two streams of data,
```
Comment (davetroiano) [inline _includes/tutorials/generate-test-data-streams/kafka/markup/dev/create-connector-02.adoc:4]: missing second stream description in the second sentence?  e.g.:

There will be two streams of data: (1) the current location of each truck, and (2) truck sensor information from each truck.

File: _data/harnesses/generate-test-data-streams/kafka.yml [hunk 15]
```diff
@@ -83,6 +83,23 @@ dev:
           render:
             file: tutorials/generate-test-data-streams/kafka/markup/dev/check-connector-02.adoc
 
+        - action: execute
```
Comment (davetroiano) [inline _data/harnesses/generate-test-data-streams/kafka.yml:86]: this causes dupe content to get rendered:

image

File: _data/harnesses/generate-test-data-streams/ksql.yml [hunk 16]
```diff
@@ -117,6 +112,9 @@ dev:
       - file: tutorial-steps/dev/create-connector-02.sql
         render:
           file: tutorials/generate-test-data-streams/ksql/markup/dev/create-connector-02.adoc
+      - file: tutorial-steps/dev/create-connector-03.sql
```
Comment (davetroiano) [inline _data/harnesses/generate-test-data-streams/ksql.yml:115]: this causes duplicate content to be rendered

image

File: _data/harnesses/generate-test-data-streams/kafka.yml (hunk 15 above)
Comment (bbejeck) [reply _data/harnesses/generate-test-data-streams/kafka.yml:86]: Fixed

File: _data/harnesses/generate-test-data-streams/ksql.yml (hunk 16 above)
Comment (bbejeck) [reply _data/harnesses/generate-test-data-streams/ksql.yml:115]: Fixed

**Review by davetroiano — APPROVED** [review]
LGTM!

**Comment by bbejeck** [comment]
Failures unrelated

**Pull request merged**